package/MDAnalysis/core/groups.py

```diff
@@ -3302,8 +3314,13 @@ def get_TopAttr(u, name, cls):
                 return attr
 
         # indices of bonds
-        b = guess_bonds(self.atoms, self.atoms.positions,
-                        vdwradii=vdwradii, box=self.dimensions)
+        b = guess_bonds(self.atoms,
+                        self.atoms.positions,
+                        vdwradii=vdwradii,
+                        box=self.dimensions,
+                        fudge_factor=fudge_factor,
+                        lower_bound=lower_bound,
+                        **kwargs)
```

Comment:
I think the question of if parameters are passed correctly mainly refers to these lines? Correct me if I'm wrong @IAlibay.

Reply:
>  we'd like to run with defaults, then check those results (for instance, the number of bonds determined) against different inputs.

I thought about that too at first but I didn't know which way is the best suit for this case.

Reply:
Can I go now for the modifications that ian mentioned above @IAlibay ? I found that `TestGuessBonds` already exists and have multiple tests for the parameters so I added additional params for the new ones to check if they passed correctly.

Reply:
Sorry about the delay, yes @ianmkenney is correct. What you want to do is check the effect of passing in the keywords instead of solely looking that they got assigned to the right things. That way if the guesser.guess_bonds changes, then you can catch that and know that something using it broke.